components/content_settings/core/browser/brave_content_settings_pref_provider_unittest.cc

```diff
@@ -729,4 +748,243 @@ TEST_F(BravePrefProviderTest, EnsureNoWildcardEntries) {
+  ShieldsFingerprintingSetting fp_settings(&provider);
+  ShieldsScriptSetting script_settings(&provider);
+  ShieldsHttpsUpgradeSetting https_upgrade_settings(&provider);
+
+  shields_cookie_settings.SetPreMigrationSettings(
+      ContentSettingsPattern::FromURL(url), CONTENT_SETTING_BLOCK);
+  cookie_settings.SetPreMigrationSettings(ContentSettingsPattern::FromURL(url),
+                                          CONTENT_SETTING_BLOCK);
+  fp_settings.SetPreMigrationSettings(ContentSettingsPattern::FromURL(url),
+                                      CONTENT_SETTING_BLOCK);
+  script_settings.SetPreMigrationSettings(ContentSettingsPattern::FromURL(url),
+                                          CONTENT_SETTING_BLOCK);
+  https_upgrade_settings.SetPreMigrationSettings(
+      ContentSettingsPattern::FromURL(url), CONTENT_SETTING_BLOCK);
+
+  testing_profile()->GetPrefs()->SetBoolean(
+      brave_shields::prefs::kAdblockAdBlockOnlyModeEnabled, true);
+
+  // Verify that adblock only mode overrides settings.
+  shields_cookie_settings.CheckSettingsWouldAllow(url);
+  cookie_settings.CheckSettingsWouldAllow(url);
+  fp_settings.CheckSettingsWouldAllow(url);
+  script_settings.CheckSettingsWouldAllow(url);
+  https_upgrade_settings.CheckSettingsWouldAsk(url);
+
+  provider.ShutdownOnUIThread();
+}
+
+TEST_F(BravePrefProviderTest,
+       AdblockOnlyModeContentSettings_DefaultValues_OffTheRecord) {
+  base::test::ScopedFeatureList feature_list;
+  feature_list.InitAndEnableFeature(brave_shields::features::kAdblockOnlyMode);
+
+  BravePrefProvider provider(
+      testing_profile()->GetPrefs(), true /* incognito */,
+      true /* store_last_modified */, false /* restore_session */);
+  GURL url("https://brave.com/");
+
+  ShieldsCookieSetting shields_cookie_settings(&provider,
+                                               testing_profile()->GetPrefs());
+  CookieSettings cookie_settings(&provider);
+  ShieldsFingerprintingSetting fp_settings(&provider);
+  ShieldsScriptSetting script_settings(&provider);
+
+  // Verify that the settings are default values.
+  shields_cookie_settings.CheckIncognitoSettings(url, CONTENT_SETTING_DEFAULT);
+  cookie_settings.CheckIncognitoSettings(url, CONTENT_SETTING_DEFAULT);
+  fp_settings.CheckIncognitoSettings(url, CONTENT_SETTING_DEFAULT);
+  script_settings.CheckIncognitoSettings(url, CONTENT_SETTING_DEFAULT);
+
+  testing_profile()->GetPrefs()->SetBoolean(
+      brave_shields::prefs::kAdblockAdBlockOnlyModeEnabled, true);
+
+  // Verify that adblock only mode overrides settings.
+  shields_cookie_settings.CheckIncognitoSettings(url, CONTENT_SETTING_ALLOW);
+  cookie_settings.CheckIncognitoSettings(url, CONTENT_SETTING_ALLOW);
+  fp_settings.CheckIncognitoSettings(url, CONTENT_SETTING_ALLOW);
+  script_settings.CheckIncognitoSettings(url, CONTENT_SETTING_ALLOW);
+
+  provider.ShutdownOnUIThread();
+}
+
+TEST_F(BravePrefProviderTest,
+       NonAdblockOnlyModeContentSettings_DefaultValues_OffTheRecord) {
```

Comment:
According to [gtest FAQ](https://github.com/google/googletest/blob/main/docs/faq.md), we probably shouldn’t use underscores in test names. I couldn’t find any explicit guidance in Chromium’s style docs.